class/practice2.2/main.cpp

```diff
@@ -0,0 +1,204 @@
+#include <string>
+#include <vector>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <ranges>
+#include <print>
+#include <cstdint>
+
+namespace {
+   constexpr const int MatrixSize = 16;
+}
+
+std::vector<std::string> splitString(const std::string& str, const char delimiter) {
+    return str  | std::ranges::views::split(delimiter)
+       | std::ranges::to<std::vector<std::string>>();
+}
+
+struct Color {
+    int r{ -1 };
+    int g{ -1 };
+    int b{ -1 };
+    };
+
+class Pixel{
+public:
+    Pixel() = default;
+    Pixel(const std::string& col){
+       auto values = splitString(col, ',');
+       if (values.size() == 3) {
+        color.r=convertToInt(values[0]);
+        color.g=convertToInt(values[1]);
+        color.b=convertToInt(values[2]);
+       }
+        
+    }
+
+    Color getColor() const {
+        return color;
+    }
+
+    std::string serToString() const{
+        std::ostringstream oss;
+        oss << (int)color.r<<","
+        <<(int)color.g<<","
+        <<(int)color.b;
+        return oss.str();
+    }
+
+    bool isValid() const {
+        return (color.r >= 0 && color.r <= 255) &&
+            (color.g >= 0 && color.g <= 255) &&
+            (color.b >= 0 && color.b <= 255);
+    }
+
+    bool compareColor(const Pixel& compare) const {
+        return (compare.color.b==color.b) && (compare.color.g==color.g) && (compare.color.r==color.r);
```

Comment:
actually this function is equivalent to comparison of pixel as color is the only data for this class. so you can use
operator== function name instead of compareColor and then you will be able to perform `pixel1 == pixel2` comparison in your code which is very short and descriptive